test/test_pty.rs

```diff
@@ -201,3 +201,33 @@ fn test_openpty_with_termios() {
     close(pty.master).unwrap();
     close(pty.slave).unwrap();
 }
+
+#[test]
+fn test_forkpty() {
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::wait;
+    // forkpty calls openpty which uses ptname(3) internally.
+    let _m0 = ::PTSNAME_MTX.lock().expect("Mutex got poisoned by another test");
+    // forkpty spawns a child process
+    let _m1 = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    let string = "naninani\n";
+    let echoed_string = "naninani\r\n";
+    let pty = forkpty(None, None).unwrap();
+    match pty.fork_result {
+        Child => {
+            write(0, string.as_bytes()).unwrap();
+            pause();  // we need the child to stay alive until the parent calls read
+        },
```

Comment:
In a Rust test, you cannot return from a forked child.  That can cause deadlocks.  Instead, you must `_exit(0)`.  The `SIGTERM` will _probably_ suffice, but I would `_exit(0)` just in case.